lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -18,18 +18,33 @@
     JSON,
     PostprocessError,
     Schema,
+    SchemaError,
     UnauthorizedAccess,
     UnsupportedAccessMode,
 )
-from pbench.server.database.models.template import Template
 from pbench.server.database.models.datasets import Dataset, DatasetNotFound
+from pbench.server.database.models.template import Template
 from pbench.server.database.models.users import User
 
 # A type defined to allow the preprocess subclass method to provide shared
 # context with the assemble and postprocess methods.
 CONTEXT = Dict[str, Any]
 
 
+class MissingBulkSchemaParameters(SchemaError):
+    """
+    The subclass schema is missing the required "controller" or dataset "name"
+    parameters required to locate a Dataset.
+    """
+
+    def __init__(self, subclass_name: str):
+        super().__init__()
+        self.subclass_name = subclass_name
+
+    def __str__(self) -> str:
+        return f"ElasticBulkBase subclass {self.subclass_name} is missing required schema parameters"
```

Comment:
Does it matter that `self.subclass_name` is an `ElasticBulkBase` subclass?  I would think that the class name would be enough to get our diagnostician started; however, if that's me, I would be awfully curious about _which_ "required schema parameters" are missing....

Reply:
I really didn't want to over-engineer this. I don't expect it to happen; and if it does, it's a major developer problem that'll be thrown the first time they try to stand up a unit test using their new subclass. How much effort is it really worth to diagnose exactly what (I) did wrong? 😀